src/plugins/ui/header_visibility_ui.ts

```diff
@@ -12,6 +21,66 @@ export class HeaderVisibilityUIPlugin extends UIPlugin {
     "isHeaderHidden",
   ] as const;
 
+  allowDispatch(cmd: Command): CommandResult {
+    switch (cmd.type) {
+      case "REMOVE_COLUMNS_ROWS":
```

Comment:
I tried to implement the fix in core earlier, but as suggested by ADRM
Here: https://github.com/odoo/o-spreadsheet/pull/2673#discussion_r1262465671
Discord: https://discord.com/channels/678381219515465750/687337979743830044/1131556525123571732

This approach makes sense since we won't be able to access the rows hidden by the UI data filter, which necessitates keeping our code within the UI.

If we don’t go beyond the issues written in specs of this task, then I believe we can skip the data filter validation part, and can ONLY handle the removal of columns/rows (via the core).

Since we have a similar file in core and (this) as an extended part in UI, I thought it would be the best place to drop CORE commands through allowDispatch. Although, in future versions (16.3 & above) this won’t work either. So, going to core would be the only choice but then again data filter has to be handled separately in UI.